src/mod_zone_difficulty_scripts.cpp

```diff
@@ -592,42 +592,77 @@ class mod_zone_difficulty_globalscript : public GlobalScript
             {
+                //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: Mythicmode is on.");
                 sZoneDifficulty->EncountersInProgress[instanceId] = GameTime::GetGameTime().count();
             }
         }
+        // Handle transition from IN_PROGRESS to DONE state
         else if (oldState == IN_PROGRESS && newState == DONE)
         {
             if (sZoneDifficulty->MythicmodeInstanceData[instanceId])
             {
-                //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: Mythicmode is on.");
-                if (sZoneDifficulty->EncountersInProgress.find(instanceId) != sZoneDifficulty->EncountersInProgress.end() && sZoneDifficulty->EncountersInProgress[instanceId] != 0)
+                //LOG_INFO("module", "\033[33mMOD-ZONE-DIFFICULTY: Mythicmode is on.\033[0m");
+
+                if (sZoneDifficulty->EncountersInProgress.find(instanceId) != sZoneDifficulty->EncountersInProgress.end() &&
+                    sZoneDifficulty->EncountersInProgress[instanceId] != 0)
                 {
                     Map::PlayerList const& PlayerList = instance->GetPlayers();
+
                     for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
                     {
                         Player* player = i->GetSource();
+
                         if (!player->IsGameMaster() && !player->IsDeveloper())
                         {
+                            //LOG_INFO("module", 
+                                     "MOD-ZONE-DIFFICULTY: Encounter completed. Map relevant. Checking for source: {}", 
+                                     player->GetGUID().GetCounter());
+                            // Log the encounter for non-GM and non-developer players
                             CharacterDatabase.Execute(
                                 "REPLACE INTO `zone_difficulty_encounter_logs` VALUES({}, {}, {}, {}, {}, {}, {})",
-                                instanceId, sZoneDifficulty->EncountersInProgress[instanceId], GameTime::GetGameTime().count(), instance->GetId(), id, player->GetGUID().GetCounter(), 64);
+                                instanceId, sZoneDifficulty->EncountersInProgress[instanceId],
+                                GameTime::GetGameTime().count(), instance->GetId(), id,
+                                player->GetGUID().GetCounter(), 64);
+                        }
+                        else
+                        {
+                            // Log the encounter for GM and developer players with an additional flag
+                            CharacterDatabase.Execute(
+                                "REPLACE INTO `zone_difficulty_encounter_logs` VALUES({}, {}, {}, {}, {}, {}, {}, 1)",
+                                instanceId, sZoneDifficulty->EncountersInProgress[instanceId],
+                                GameTime::GetGameTime().count(), instance->GetId(), id,
+                                player->GetGUID().GetCounter(), 64);
```

Comment:
64 isn't required here. There's one less placeholder than before.
1 seems like a bad choice for the mode to indicate a GM was present. 1 and 64 are currently taken for "normal" and "mythic".
Let's reserve low numbers for players and start from the back, using high numbers with GMs.

Reply:
This is actually an additional column in the table. 0 for players (default db value) and 1 for gm. The `mode` column remains as is.

Would you like me to remove this code, or add the column to sql?